packages/smooth_app/lib/pages/scan/scan_page.dart

```diff
@@ -1,48 +1,47 @@
 import 'package:flutter/material.dart';
-import 'package:flutter_svg/flutter_svg.dart';
 import 'package:provider/provider.dart';
 import 'package:smooth_app/data_models/continuous_scan_model.dart';
 import 'package:smooth_app/database/local_database.dart';
 import 'package:smooth_app/database/product_query.dart';
 import 'package:smooth_app/pages/scan/continuous_scan_page.dart';
 
-class ScanPage extends StatelessWidget {
+class ScanPage extends StatefulWidget {
   const ScanPage();
 
   @override
-  Widget build(BuildContext context) {
+  State<ScanPage> createState() => _ScanPageState();
+}
+
+class _ScanPageState extends State<ScanPage> {
+  ContinuousScanModel? _model;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _updateModel();
+  }
+
+  Future<void> _updateModel() async {
     final LocalDatabase localDatabase = context.watch<LocalDatabase>();
-    return FutureBuilder<ContinuousScanModel?>(
-        future: ContinuousScanModel(
-          languageCode: ProductQuery.getCurrentLanguageCode(context),
-          countryCode: ProductQuery.getCurrentCountryCode(),
-        ).load(localDatabase),
-        builder: (BuildContext context,
-            AsyncSnapshot<ContinuousScanModel?> snapshot) {
-          if (snapshot.connectionState == ConnectionState.done) {
-            final ContinuousScanModel? continuousScanModel = snapshot.data;
-            if (continuousScanModel != null) {
-              return ContinuousScanPage(continuousScanModel);
-            }
-          }
-          return const Center(child: CircularProgressIndicator());
-        });
+    if (_model == null) {
+      _model = await ContinuousScanModel(
+        languageCode: ProductQuery.getCurrentLanguageCode(context),
+        countryCode: ProductQuery.getCurrentCountryCode(),
+      ).load(localDatabase);
+    } else {
+      await _model?.refresh();
+    }
+    setState(() {});
   }
 
-  static Widget getHero(final Size screenSize) => Hero(
-        tag: 'action_button',
-        child: Container(
-          width: screenSize.width,
-          height: screenSize.height,
-          color: Colors.black,
-          child: Center(
-            child: SvgPicture.asset(
-              'assets/actions/scanner_alt_2.svg',
-              width: 60.0,
-              height: 60.0,
-              color: Colors.white,
-            ),
-          ),
-        ),
-      );
+  @override
+  Widget build(BuildContext context) {
+    if (_model == null) {
+      return const Center(child: CircularProgressIndicator());
+    }
+    return ChangeNotifierProvider<ContinuousScanModel>(
+      create: (BuildContext context) => _model!,
```

Comment:
what's the norm for unused parameters in Flutter ? I've seen "_" in such cases, something like this:

create: (_) => _model!,

Reply:
The style guide indeed recommends `_`, `__`, etc for unused parameters of anonymous local functions:
https://dart.dev/guides/language/effective-dart/style#prefer-using-_-__-etc-for-unused-callback-parameters